lib/active_merchant/billing/gateways/barclaycard_smartpay.rb

```diff
@@ -60,6 +60,10 @@ def credit(money, creditcard, options = {})
         post = payment_request(money, options)
         post[:amount] = amount_hash(money, options[:currency])
         post[:card] = credit_card_hash(creditcard)
+        post[:dateOfBirth] = options[:date_of_birth] if options[:date_of_birth]
+        post[:entityType]  = options[:entity_type] if options[:entity_type]
+        post[:nationality] = options[:nationality] if options[:nationality]
+        post[:shopperName] = options[:shopper_name] if options[:shopper_name]
```

Comment:
Now without the `requires!` we can at least be helpful with a fallback like `|| creditcard.name`

Reply:
I wrote this, but as I started updating the tests to handle it, I realized I disagree with the feature: for us to do this, you'd have to 1) issue a credit, 2) successfully supply the date of birth, entity type, and nationality, but 3) forget shopper name. That's already an edge case of an edge case, but yoinking data from the credit card fields has a fun side-effect that the user may get the credit rejected for _having the wrong shopper name_, and then will be trying to figure out how that's even possible/what the shopper name was in the first place.

I thus think this is probably the best path forward as-is.

Reply:
I think that's fair 👍